src/graph.jl

```diff
@@ -15,13 +15,92 @@ struct BaseGraph<:AbstractGraph
 end
 
 
+function polygon_array(coords::Vector{Vector{Vector{Vector{Float64}}}})::Array{LibGEOS.Polygon}
+    # Takes a complex vector object and returns an array of Polygons
+    # TODO(matthew): implement this, and add return type to signature
+    return []
+end
+
+
+function min_bounding_rect(coords::Vector{Vector{Vector{Vector{Float64}}}})::Tuple
+    # Takes a complex vector object and returns an (axis-aligned) minimum
+    # bounding rectangle
+    # TODO(matthew): implement this, and add return type to signature
+    return ()
+end
+
+
+function read_table(filepath::AbstractString)::Shapefile.Table
+    """ Read table from shapefile. If a .shp and a .dbf file of the same name
+        are not found, then we throw an error.
+    """
+    prefix = splitext(filepath)[1]
+    if !isfile(prefix * ".shp") || !isfile(prefix * ".dbf")
```

Comment:
See my comments in #31 about file extensions. I'd rather this only fail if `Shapefile.Table(prefix)` doesn't work. This is in keeping with the [EAFP pattern](https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/) commonly used in Python, which I tend to favor for things like file I/O.

Reply:
The issue is that `Shapefile.Table(prefix)` may work even if the `.dbf` doesn't exist. For us, the requirement that there be a `.shp` and a `.dbf` of the same name exists because we need **both** geometries and attributes. I'm going to keep this in here for now but will be sure to also check for `.SHP` and `.DBF` to make this a little more robust.